### DISCUSSION TITLE
Number of epochs per fold?

### DISCUSSION THREAD

**Discussion by paul-reiners** [MIC-DKFZ/nnUNet#626] [Q&A] [answered]
How many epochs per fold are run?

**Comment by FabianIsensee** [comment, marked as answer]
1000

**Comment by alaurent4** [comment]
Hello, seems relevant to ask my question here if I may.
Im using the 3D full config. The 5 folds have been running for more than 2 days than and are far from 1000 epochs. 
Is there a way to do some prediction even if the models are not finished yet ?
Or is it possible to reduce the number of epochs ?

**Reply by FabianIsensee** [reply]
read this and run the benchmark: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/expected_epoch_times.md

It could be that your system is underperforming.

You can use model_latest and model_best checkpoints to predict (see `nnUNet_predict -h`) but it is not recommended!

**Reply by alaurent4** [reply]
Thanks for your help! I used -chk model_best , seems pretty fair to me.

**Reply by FabianIsensee** [reply]
fair? yes. Best possible? No
:-)

**Reply by alaurent4** [reply]
But if the loss is not decreasing anymore after 100 epochs on each fold, and I pick any 'best_model' after that point, it should match with the checkpoint right ?

**Reply by FabianIsensee** [reply]
trust me when I say it is best to let it run ^^. The learning rate decreases towards the end of the training and on some (!) datasets the scores improve a little bit. Training nnU-Net does not take long (~2 days on 2080ti and that is the slowest GPU. It's 16h on A100). Just let it do its thing :-)

**Reply by alaurent4** [reply]
Okay :)

**Comment by paul-reiners** [comment]
Can the five folds be run concurrently?

**Reply by FabianIsensee** [reply]
yes. Let fold0 start training (GPU utilization is high). Then start the other folds. Waiting for fold0 to start is necessary because it will unpack the training data first and unpacking should not be done by multiple trainings simultaneously
